src/variables.c

```diff
@@ -419,7 +419,7 @@ uzbl_variables_dump_events ()
 void
 variable_free (UzblVariable *variable)
 {
-    if ((variable->type == TYPE_STR) && variable->value.s) {
+    if ((variable->type == TYPE_STR) && variable->value.s && variable->writeable) {
         g_free (*variable->value.s);
         if (!variable->builtin) {
```

Comment:
Ah, this is what was saving us from double frees before I think. I'd certainly see them since I set `MALLOC_CHECK_=3` which aborts on double free and other memory error problems.